### PR TITLE
Add support for single-node Kubernetes deploys on EC2

### PR DESCRIPTION
**Pull request by djwhatle**
**Changes**
 - Fixed Kubernetes repo file
 - Fixed import of Kubernetes repo GPG key
 - Added "wait" for service-catalog readiness that waits for actual readiness to solve problem with ClusterServiceBroker creation
 - Added bash-completion package and enabled completion for `kubectl`
 - Add Kubernetes security group
 - Disabled `docker-storage-setup` in Kubernetes env config steps (otherwise service-catalog fails to come online)
 - Made `become` usage more granular so that top-level `become: true` could be removed
 - Improve naming of existing security groups
 - Update AMI IDs to RHEL 7.4

**How to use**

- Unset `hostname` in `config/my_vars.yml` in order to get SSL working
- `ec2/minimal/terminate_instance.sh` to terminate any existing ec2 infrastructure created by catasb
- `ec2/minimal/run_create_infrastructure.sh -k` to create infrastructure with k8s appropriate infra (ports opened)
- `ec2/minimal/run_setup_environment.sh -k` to setup the AWS environment with a single-node Kubernetes environment
- `ec2/minimal/reset_environment.sh -k` to setup the AWS environment with a single-node Kubernetes environment
- `ec2/minimal/terminate_instance.sh -k` to terminate the AWS instance created for k8s

### CODE REVIEW

**Comment by johnkim76** [comment]
VISIACK, but will approve after getting a successful run

**Comment by johnkim76** [comment]
@djwhatle would it be ok if we add a 'force: yes' for this git clone task [here](https://github.com/djwhatle/catasb/blob/more-kubernetes-support/ansible/roles/ssl_setup/tasks/main.yml#L19)? The task fails when you try to re-run the playbook b/c the repo already exists.

**Comment by johnkim76** [comment]
@djwhatle it looks like we need to add 'remote_src: yes' [here](https://github.com/djwhatle/catasb/blob/more-kubernetes-support/ansible/roles/kubernetes_setup/tasks/setup_helm.yml#L35) for the 'Untarring helm' task, and [here](https://github.com/djwhatle/catasb/blob/more-kubernetes-support/ansible/roles/kubernetes_setup/tasks/setup_helm.yml#L40) for the 'Copying helm binary...' task.  Here's the reference [link](https://github.com/ansible/ansible/issues/32766#issuecomment-345148707)

File: ansible/roles/aws_docker_setup/tasks/minimal.yml [hunk 1]
```diff
@@ -1,48 +1,51 @@
...
+    - name: Created docker group
+      group:
+        name: docker
+        state: present
 
-  - name: Added ec2-user to group
-    user:
-      name: ec2-user
-      groups: "ec2-user,adm,wheel,systemd-journal,docker"
+    - name: Added ec2-user to group
```
Comment (jwmatthews) [inline ansible/roles/aws_docker_setup/tasks/minimal.yml]: Just checking if this formatting change is intentional.

File: ansible/roles/aws_ebs_volumes/tasks/main.yml [hunk 2]
```diff
@@ -1,54 +1,57 @@
...
-    poll: 0
-    filesystem:
-      fstype: ext4
-      dev: '{{ item }}'
-      opts: -E lazy_itable_init
-    with_items:
-      - "{{ var_lib_docker_block_device }}"
-      - "{{ tmp_block_device }}"
-      - "{{ persistent_vol_block_device }}"
+    - name: Create ext4 filesystems for /var/lib/docker, /tmp, persistent volumes
```
Comment (jwmatthews) [inline ansible/roles/aws_ebs_volumes/tasks/main.yml]: Same here, is the formatting change intentional?

File: ansible/roles/aws_docker_setup/tasks/minimal.yml (hunk 1 above)
Comment (djwhatle) [reply ansible/roles/aws_docker_setup/tasks/minimal.yml]: Do you mean the tabbing out? I added several "become: true" blocks to replace a play-wide "become: true" that used to exist on the AWS setup tasks. Settings "become: true" for the entire play wasn't working nicely with the k8s automation.

File: ansible/roles/aws_ebs_volumes/tasks/main.yml (hunk 2 above)
Comment (djwhatle) [reply ansible/roles/aws_ebs_volumes/tasks/main.yml]: If referring to the tabbing out, see comment above.

**Comment by djwhatle** [comment]
Just rebased this on the latest 3.9 changes.

Tests run (will update as I go along):
```
=== Local ===

[ OK ] - Deploy local OpenShift with no changes to config file
[  ] - Deploy local OpenShift with "3.9" set for versions in config file

[  ] - Deploy local Kubernetes

=== AWS ===

[  ] - Deploy AWS OpenShift with no changes to config file
[  ] - Deploy AWS OpenShift with "3.9" set in config file
[  ] - Deploy AWS OpenShift with "3.9" and "ec2_use_proxy" set in config file

[  ] - Deploy AWS Kubernetes

**Comment by djwhatle** [comment]
Now seeing "OpenShift web console is not available", "The application is currently not serving requests. It might not be installed or is still installing." when testing latest OpenShift. Need to check if this matches current conditions on master.

**Comment by djwhatle** [comment]
I was able to verify that the Web Console issue is _not_ present on master, only on this branch. Will need to investigate.

**Comment by djwhatle** [comment]
Tested against Jenkins. Appears to be solid.

**Comment by johnkim76** [comment]
successfully tested

**Pull request merged**